usr.sbin/installboot/arch/evbarm.c

```diff
@@ -0,0 +1,247 @@
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if HAVE_NBTOOL_CONFIG_H
+#include "nbtool_config.h"
+#endif
+
+#include <sys/cdefs.h>
+#if !defined(__lint)
+__RCSID("$NetBSD$");
+#endif  /* !__lint */
+
+#include <stdio.h>
+#include <string.h>
+#include <limits.h>
+
+#include "installboot.h"
+#include "evboards.h"
+
+static int	evbarm_setboot(ib_params *);
+static int	evbarm_clearboot(ib_params *);
+static int	evbarm_editboot(ib_params *);
+
+struct ib_mach ib_mach_evbarm = {
+	.name		=	"evbarm",
+	.setboot	=	evbarm_setboot,
+	.clearboot	=	evbarm_clearboot,
+	.editboot	=	evbarm_editboot,
+	.valid_flags	=	IB_BOARD | IB_SOC,
+	.mach_flags	=	MF_UBOOT,
+};
+
+static int meson_gxbb_setboot(ib_params *);
+
+static const struct evboard_methods meson_gxbb_methods = {
+	.name		=	"meson-gxbb",
+	.setboot	=	meson_gxbb_setboot,
+};
+
+static int meson_gxl_setboot(ib_params *);
+
+static const struct evboard_methods meson_gxl_methods = {
+	.name		=	"meson-gxl",
+	.setboot	=	meson_gxl_setboot,
+};
+
+static int rk3328_setboot(ib_params *);
+
+static const struct evboard_methods rk3328_methods = {
+	.name		=	"rk3328",
+	.setboot	=	rk3328_setboot,
+};
+
+static int rk3399_setboot(ib_params *);
+
+static const struct evboard_methods rk3399_methods = {
+	.name		=	"rk3399",
+	.setboot	=	rk3399_setboot,
+};
+
+static int sunxi_setboot(ib_params *);
+
+static const struct evboard_methods sunxi_methods = {
+	.name		=	"sunxi",
+	.setboot	=	sunxi_setboot,
+};
+
+static const struct evboard_methods * const evbarm_methods[] = {
+	&meson_gxbb_methods,
+	&meson_gxl_methods,
+	&rk3328_methods,
+	&rk3399_methods,
+	&sunxi_methods,
+	NULL,
+};
+
+static int
+evbarm_setboot(ib_params *params)
+{
+	const struct evboard_methods *m;
+
+	m = evb_methods_lookup(params, evbarm_methods);
+	if (m == NULL)
+		return 0;
+
+	return m->setboot(params);
+}
+
+static int
+evbarm_clearboot(ib_params *params)
+{
+	const struct evboard_methods *m;
+
+	m = evb_methods_lookup(params, evbarm_methods);
+	if (m == NULL)
+		return 0;
+
+	if (m->clearboot != NULL)
+		return m->clearboot(params);
+
+	return no_clearboot(params);
+}
+
+static int
+evbarm_editboot(ib_params *params)
+{
+	const struct evboard_methods *m;
+
+	m = evb_methods_lookup(params, evbarm_methods);
+	if (m == NULL)
+		return 0;
+
+	if (m->editboot != NULL)
+		return m->editboot(params);
+
+	return no_editboot(params);
+}
+
+/*
+ * meson-gxbb methods
+ */
+static const struct evboard_uboot_desc meson_gxbb_uboot_description[] = {
+	{ .filename = "bl1.bin.hardkernel",
```

Comment:
The `.hardkernel` suffix is specific to the ODROID-C2 (vendor name in this case is Hardkernel).

Reply:
Yes, but it's what your image building scripts use.  There don't appear to be any other meson-gxbb examples for me to look at.